### PR TITLE
Permissiveness questions & inherited properties

### PR DESCRIPTION
**Pull request by rufflian**
- Added Permissiveness questions for Build, Update, and Child Propagation
- Added support for inherited properties and updated relevant steps
- UI fixes for multi-question steps
- Minor performance and type fixes

Below shows the child propagation input for an item class wizard inheriting from a parent that has overridden this property
childpropagation

Below shows the interface for an item class wizard inheriting from a parent (not overridden) and changing the values
inheritancealert

### CODE REVIEW

**Review by long-banana — APPROVED** [review]
Sorry for the delay on this. Looks great!

File: src/createItemClass/UpdatePermissiveness.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                ?.updatePermissiveness as (keyof State.AnchorPermissivenessType)[]
+            }
+            parentItemClass={parentItemClass}
+            error={props.errors.updatePermissiveness}
+          />
+          {props.values.updatePermissiveness?.length > 1 && (
+            <RadioInput.Inline
+              name="permissivenessToUse"
+              title="Which permissiveness should be used to create this item class?"
+              help="How do you know what to put here //QUESTION"
```
Comment (rufflian) [inline src/createItemClass/UpdatePermissiveness.tsx]: @long-banana how does the user know what to select for the updatePermissivenessToUse? Does it need to be one of the selected update permissivenesses?

Comment (long-banana) [reply src/createItemClass/UpdatePermissiveness.tsx]: It should be whatever permissiveness they want to assign for updating the item class config as a whole. Assuming I'm correct in understanding the context this radio is shown in.

Comment (rufflian) [reply src/createItemClass/UpdatePermissiveness.tsx]: Here: https://github.com/raindrops-protocol/raindrops/blob/02fb16c4e96a68dd9c456b55701f0fb81f785761/rust/item/src/lib.rs#L354 it seems to be related to the parent? So I guess this should be limited to parent update permissiveness options? Can we leave out for item classes without parents?

Comment (long-banana) [reply src/createItemClass/UpdatePermissiveness.tsx]: Yes, we can leave out if there is no parent.

Comment (rufflian) [reply src/createItemClass/UpdatePermissiveness.tsx]: Updated in https://github.com/raindrop-studios/creator-ui/pull/8/commits/9fef2d87de08a29eca646ca4b810946bbae8993d

**Pull request merged**